cannot compile on windows: expected u32, got i32 in gdnative-sys

error is about failing to case an `i32` as an `u32`.

It seems related to #247 

My latest successful build on windows was with commit [e45f4ff](https://github.com/GodotNativeTools/godot-rust/commit/e45f4ffd3232ee24cb5bda791f369118cbfd1cb5)

```
   Compiling gdnative-derive v0.7.0 (https://github.com/GodotNativeTools/godot-rust#337a3dd5)
   Compiling serde_json v1.0.47
   Compiling gdnative_bindings_generator v0.7.0 (https://github.com/GodotNativeTools/godot-rust#337a3dd5)
   Compiling gdnative-sys v0.7.0 (https://github.com/GodotNativeTools/godot-rust#337a3dd5)
   Compiling gdnative-core v0.7.0 (https://github.com/GodotNativeTools/godot-rust#337a3dd5)
   Compiling gdnative-bindings v0.7.0 (https://github.com/GodotNativeTools/godot-rust#337a3dd5)
error[E0308]: mismatched types
  --> C:\Users\runneradmin\.cargo\git\checkouts\godot-rust-6b1e607c51143c56\337a3dd\gdnative-sys\src\lib.rs:17:24
   |
17 |     if (*api).type_ == api_type {
   |                        ^^^^^^^^ expected `u32`, found `i32`
   |
help: you can convert an `i32` to `u32` and panic if the converted value wouldn't fit
   |
17 |     if (*api).type_ == api_type.try_into().unwrap() {
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> C:\Users\runneradmin\.cargo\git\checkouts\godot-rust-6b1e607c51143c56\337a3dd\gdnative-sys\src\lib.rs:23:36
   |
23 |                 && (*api).type_ == api_type
   |                                    ^^^^^^^^ expected `u32`, found `i32`
   |
help: you can convert an `i32` to `u32` and panic if the converted value wouldn't fit
   |
23 |                 && (*api).type_ == api_type.try_into().unwrap()
   |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> C:\Users\runneradmin\.cargo\git\checkouts\godot-rust-6b1e607c51143c56\337a3dd\gdnative-sys\src\lib.rs:33:34
   |
33 |         if (*extension).type_ == api_type {
   |                                  ^^^^^^^^ expected `u32`, found `i32`
   |
help: you can convert an `i32` to `u32` and panic if the converted value wouldn't fit
   |
33 |         if (*extension).type_ == api_type.try_into().unwrap() {
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> C:\Users\runneradmin\.cargo\git\checkouts\godot-rust-6b1e607c51143c56\337a3dd\gdnative-sys\src\lib.rs:39:46
   |
39 |                     && (*extension).type_ == api_type
   |                                              ^^^^^^^^ expected `u32`, found `i32`
   |
help: you can convert an `i32` to `u32` and panic if the converted value wouldn't fit
   |
39 |                     && (*extension).type_ == api_type.try_into().unwrap()
   |                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `gdnative-sys`.
```

--- a/gdnative-sys/build.rs
+++ b/gdnative-sys/build.rs
@@ -138,6 +138,16 @@ mod api_wrapper {
             let mut result = Vec::new();
             result.extend(self.core.nexts());
             for extension in &self.extensions {
+                // Workaround: Interface for the ANDROID extension is broken in Godot 3.2,
+                // making it impossible to support the extension in a forward compatible manner.
+                // The extension is disabled until the compatibility policy is figured out in
+                // upstream.
+                //
+                // Related issue: #296
+                if extension.type_ == "ANDROID" {
+                    continue;
+                }
+
                 result.extend(extension.nexts());
             }
             result

--- a/gdnative-sys/src/lib.rs
+++ b/gdnative-sys/src/lib.rs
@@ -14,13 +14,13 @@ unsafe fn find_api_ptr(
     version_minor: u32,
 ) -> *const godot_gdnative_api_struct {
     let mut api = core_api as *const godot_gdnative_api_struct;
-    if (*api).type_ == api_type {
+    if (*api).type_ as u32 == api_type as u32 {
         while !api.is_null() {
             if (*api).version.minor == version_minor
                 && (*api).version.major == version_major
                 // The boolean expression below SHOULD always be true;
                 // we will double check to be safe.
-                && (*api).type_ == api_type
+                && (*api).type_ as u32 == api_type as u32
             {
                 return api;
             }
@@ -30,13 +30,13 @@ unsafe fn find_api_ptr(
     for i in 0..(*core_api).num_extensions {
         let mut extension =
             *(*core_api).extensions.offset(i as _) as *const godot_gdnative_api_struct;
-        if (*extension).type_ == api_type {
+        if (*extension).type_ as u32 == api_type as u32 {
             while !extension.is_null() {
                 if (*extension).version.minor == version_minor
                     && (*extension).version.major == version_major
                     // The boolean expression below SHOULD always be true;
                     // we will double check to be safe.
-                    && (*extension).type_ == api_type
+                    && (*extension).type_ as u32 == api_type as u32
                 {
                     return extension;
                 }
@@ -45,7 +45,8 @@ unsafe fn find_api_ptr(
         }
     }
     panic!(
-        "Couldn't find API: {:?}",
-        (api_type, version_major, version_minor)
+        "Couldn't find API struct with type {type}, version {version:?}",
+        type = api_type,
+        version = (version_major, version_minor),
     );
 }

